---
title: '01 - Seurat Data Import'
author: "Ashley B. Robbins"
date: "`r Sys.Date()`"
output: 
  html_document:
    code_folding: "show"
    toc: true
    toc_float: true
    theme: 
      bootswatch: yeti
params:
  dir: "/Users/robbinsa1/Library/CloudStorage/Box-Box/07192022/counts/Plate_1/"
  meta: "/Users/robbinsa1/Library/CloudStorage/Box-Box/07192022/HD_Plate1_metadata.csv"
  experiment_id: "HDSPLiTseq_Plate1"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
knitr::opts_knit$set(root.dir = params$dir)
```

# Goals

1.  Import counts.tsv output of the [SPLiT-seq demultiplexing](https://github.com/paulranum11/SPLiT-Seq_demultiplexing) tool for each sub-library for a given experiment
    -   *In a SPLiTseq experiment nuclei/cells are barcoded for single-cell sequencing via a series of split-pool reactions with barcoded oligos. At the end of SPLiTseq barcoding all nuclei/cells are pooled into a single tube and split based on a desired \# of cells per sub-library (\~500 - 10K nuclei/cells per sub-library).*
2.  Create a list object of all counts matrices and use lapply functions to create [Seurat](https://satijalab.org/seurat/) objects for each sub-library.
3.  Integrate metadata with Seurat objects.
    -   *During the first round of barcoding in the SPLiTseq protocol nuclei/cells are loaded into a 96 well plate. Each well of this plate corresponds to a single R1 barcode. Multiplexed samples can be loaded into separate wells of the R1 barcoding plate and the R1 barcode can be used as a sample identifier to later match with known sample information (Age, Genotype, Treatment, etc).*
4.  Merge all individual sub-library Seurat objects into a single experiment Seurat object.
5.  **Optional:** Split the merged Seurat object based on anatomical region for later filtering and region-specific cell-type annotation.
    -   *Due to the ability to multiplex a large number of samples and experimental conditions in a single SPLiTseq experiment there may often be sub-libraries that contain nuclei/cells from multiple brain and/or peripheral regions. In order to best identify cell-types for a given region we want to work with a Seurat object with cells that originate from a single anatomical region at a time.*

    -   *In order to split objects by region identity, the region origin of a given sample must be reflected in the metadata.*

***Check out these resources for more information on the SPLiTseq methodology:***

-   Original paper: <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7643870/>

-   Detailed description of chemistry: <https://teichlab.github.io/scg_lib_structs/methods_html/SPLiT-seq.html>

# Set up R environment

## Load necessary packages

```{r load packages}
suppressPackageStartupMessages({
  library(Seurat)
  library(readr)
  library(tidyseurat)
  library(dplyr)
  library(tidyverse)
  library(details)
})
```

## Write functions

**import_counts(x) :** read in counts matrices as data.frames with column for gene ID. Input is a tsv formatted counts matrix.

```{r import_counts}
import_counts <- function(x){
  as.data.frame(read_delim(x,"\t", escape_double = FALSE, trim_ws = TRUE))%>%
    column_to_rownames(var = "gene")
}
```

**add_barcode(x):** extract the first 1-8 bp of cell barcode as R1 barcode to later link with corresponding sample information. Input is a seurat object.

```{r add_barcode }
add_barcode<- function(x){
  CellsMeta <- x@meta.data
  Barcode <- substr(rownames(CellsMeta), 1, 8)
  x <- AddMetaData(object = x, metadata = Barcode, col.name = "Barcode")
}
```

**add_meta(x,y)**: join metadata columns to seurat object based on the R1 barcode sequence. Input is a seurat object *(x)* and metadata dataframe *(y)*.

```{r add_meta}
add_meta <- function(x, y){
  left_join(x, y, by = "Barcode", .keep_all = FALSE)
}
```

# Import data into Seurat

### Step 1: Read in the corresponding metadata for this experiment

```{r sample metadata, results = "hide"}
## Read in metadata
metadata <- read_csv(params$meta)
```

**View metadata information**

```{r view metadata}
head(as_tibble(metadata),10)
```

### Step 2: Create a list object containing counts files

```{r get counts files}
file.exists(params$dir) #confirm your directory path exists
files <- as.list(list.files(params$dir, pattern = "*counts.tsv.gz")) #create list object containing only tsv counts files
```

### Step 3: Import counts matrices with lapply

```{r import counts, message = FALSE}
libID <- sub('(.*)_counts.tsv.gz.*','\\1',files) #pull sublibrary IDs from counts files to use as names for the list object
libID

Sys.setenv("VROOM_CONNECTION_SIZE" = 5000000)
counts <- lapply(files,import_counts)
names(counts) <- libID
```

### Step 4: Load counts into Seurat separated by sublibrary

```{r load Seurat}
seurat.list = list()

for (i in 1:length(counts)) {
  seurat.list[[i]] <- CreateSeuratObject(counts[[i]], min.cells = 10, min.features = 100, project = libID[[i]]);
  seurat.list[[i]][["Sublibrary"]] = libID[[i]]
}
## View seurat objects
seurat.list
```

### Step 5: Add metadata information to each Seurat object

```{r add metadata Seurat}
seurat.list <- lapply(seurat.list, add_barcode)# add barcode information to metadata
seurat.list <- lapply(seurat.list, add_meta, y = metadata)# merge seurat@meta.data with additional metadata via barcode slot 
```

### Step 6: Merge Seurat objects and save to disk

```{r Seurat merge}
seurat_merged  <- merge(seurat.list[[1]],y = seurat.list[2:length(seurat.list)], 
                    add.cell.ids = libID, project = params$experiment_id)
seurat_merged

saveRDS(seurat_merged, file = paste("seurat_merged_",params$experiment_id,".rds", sep = ""))

# Save merged seurat object
```

### Step 7: (OPTIONAL) Split by Region and save to disk

```{r Seurat split}
## Change active identity to "Region"
Idents(seurat_merged) <- "Region"
table(Idents(seurat_merged))

## Subset by region
###Striatum
seurat_striatum <- subset(seurat_merged, idents = "Striatum")
seurat_striatum
saveRDS(seurat_striatum, file = paste("seurat_striatum_",params$experiment_id,".rds", sep = ""))

###Cortex
seurat_cortex <- subset(seurat_merged, idents = "Cortex")
seurat_cortex
saveRDS(seurat_cortex, file = paste("seurat_cortex_",params$experiment_id,".rds", sep = ""))

###Heart
seurat_heart <- subset(seurat_merged, idents = "Heart")
seurat_heart
saveRDS(seurat_heart, file = paste("seurat_heart_",params$experiment_id,".rds", sep = ""))
```

# Session Info

```{r session info, echo=FALSE}
sessioninfo::session_info()%>%
  details::details(
    summary = 'Current session info',
    open    = TRUE
  )
```
